m2e-maven-runtime/org.eclipse.m2e.maven.indexer/pom.xml

```diff
@@ -50,10 +54,34 @@
       <version>0.3.4</version>
       <scope>provided</scope>
     </dependency>
+    <dependency>
+      <groupId>org.apache.maven.archetype</groupId>
+      <artifactId>archetype-common</artifactId>
+      <version>${archetype-common.version}</version>
+      <scope>provided</scope>
+    </dependency>
   </dependencies>
 
   <build>
     <plugins>
+    <!-- Need to regenerate the "javax.inject.Named" index to include additional 
+        injectable classes that do not override those provided by maven-indexer, 
+        this is included in the classpath in front of maven-indexer so it takes precedence 
+        over the one in the maven-indexer jar -->
```

Comment:
"javax.inject.Named" is now generated, but here is the reason why it exists at all